cluster-install/install.sh

```diff
@@ -1115,13 +1139,24 @@ apply_kubectl_patches() {
 wait_for_deployment_running() {
     local -r deployment="$1"
     local -r namespace="${2:-"default"}"
-    local -r timeout="${3:-"2m"}"
+    local -r timeout="${3:-"2"}" # Minutes
     [ -z "$deployment" ] && fatal "no deployment name given"
 
+    local -r sleep_time="5" # Seconds
+    local -r max_checks="$(( (timeout * 60) / sleep_time ))"
+    local counter=0
     loginfo "Waiting on deployment/$deployment (timeout: $timeout)..."
-    if kubectl rollout status -n "$namespace" "deploy/$deployment" --timeout="$timeout" &> /dev/null; then
-        return 0
-    fi
+    while ((counter < max_checks)); do
+        if kubectl rollout status -n "$namespace" "deploy/$deployment" -w=false &> /dev/null; then
```

Comment:
After testing with just using kubectl rollout's `--timeout` flag, I realized the command will exit early when the deployment in certain failure states, which made the deployment monitoring exit early in many cases since we generally break a few eggs in the process of installing/upgrading things. 

Not sure if there's another option to kubectl rollout that makes it more tolerant, but for now I reverted to using a `while` loop.